services/datamanager/datasync/sync.go

```diff
@@ -69,11 +69,18 @@ type syncer struct {
 }
 
 // ManagerConstructor is a function for building a Manager.
-type ManagerConstructor func(identity string, client v1.DataSyncServiceClient, logger logging.Logger, captureDir string) (Manager, error)
+type ManagerConstructor func(identity string, client v1.DataSyncServiceClient, logger logging.Logger,
+	captureDir string, maxSyncThreadsConfig int) (Manager, error)
 
 // NewManager returns a new syncer.
-func NewManager(identity string, client v1.DataSyncServiceClient, logger logging.Logger, captureDir string) (Manager, error) {
+func NewManager(identity string, client v1.DataSyncServiceClient, logger logging.Logger,
+	captureDir string, maxSyncThreadsConfig int,
```

Comment:
I'd rather keep any logic about the config (and conditioning on the config) constrained to `builtin.go`, and make this operate independently with whatever parameters are passed in.

This is similar to the capture directory, which gets a default value if it's empty and only resets collectors if it changes.

Currently the code is [here](https://github.com/viamrobotics/rdk/blob/d3f3bff53f8f5b07db02f962605d192fbec4eaa7/services/datamanager/builtin/builtin.go#L429-L434), which I'd mimic (except also fix that linked code to be an if-else clause..otherwise oof that's ugly)

Reply:
lol, makes sense and will fix!